cloud/google/gce_net.py

```diff
@@ -221,12 +230,22 @@ def main():
 
         # user wants to create a new network that doesn't yet exist
         if name and not network:
-            if not ipv4_range:
-                module.fail_json(msg="Network '" + name + "' is not found. To create network, 'ipv4_range' parameter is required",
+            if not ipv4_range and mode == 'legacy':
+                module.fail_json(msg="Network '" + name + "' is not found. To create network in legacy mode, 'ipv4_range' parameter is required",
                     changed=False)
 
             try:
-                network = gce.ex_create_network(name, ipv4_range)
+                if mode != 'legacy':
+                    import inspect
+
+                    args = inspect.getargspec(gce.ex_create_network)
```

Comment:
Could you do the version check[ like this](https://github.com/tgbyte/ansible-modules-core/blob/5be9f4955136ed0679144c8e99e7cc129a3da7ef/cloud/google/gce.py#L598)?  The argument to that method is in v1.0 and is expected to remain.